soco/events.py

```diff
@@ -388,261 +212,152 @@ def __init__(self, service, event_queue=None):
+            url (str): The full endpoint to which the request is being sent.
+            headers (dict): A dict of headers, each key and each value being
+                of type `str`.
+            success (function): An optional function to be called if the
+                request succeeds. The function will be called with a dict
+                of response headers as its only parameter.
+            failure (function): An optional function to call if the request
+                fails. The function will be called without any parameter.
+
         """
-        # NB This code is sometimes called from a separate thread (when
-        # subscriptions are auto-renewed. Be careful to ensure thread-safety
-
-        if self._has_been_unsubscribed:
-            raise SoCoException(
-                'Cannot renew subscription once unsubscribed')
-        if not self.is_subscribed:
-            raise SoCoException(
-                'Cannot renew subscription before subscribing')
-        if self.time_left == 0:
-            raise SoCoException(
-                'Cannot renew subscription after expiry')
-
-        # SUBSCRIBE publisher path HTTP/1.1
-        # HOST: publisher host:publisher port
-        # SID: uuid:subscription UUID
-        # TIMEOUT: Second-requested subscription duration (optional)
-        headers = {
-            'SID': self.sid
-        }
-        if requested_timeout is None:
-            requested_timeout = self.requested_timeout
-        if requested_timeout is not None:
-            headers["TIMEOUT"] = "Second-{}".format(requested_timeout)
-        response = requests.request(
-            'SUBSCRIBE',
-            self.service.base_url + self.service.event_subscription_url,
-            headers=headers)
-        response.raise_for_status()
-        timeout = response.headers['timeout']
-        # According to the spec, timeout can be "infinite" or "second-123"
-        # where 123 is a number of seconds.  Sonos uses "Second-123" (with a
-        # a capital letter)
-        if timeout.lower() == 'infinite':
-            self.timeout = None
+        try:
+            response = requests.request(method, url, headers=headers)
+            response.raise_for_status()
+        except requests.exceptions.RequestException:
+            log.exception('Problem sending request')
         else:
-            self.timeout = int(timeout.lstrip('Second-'))
-        self._timestamp = time.time()
-        self.is_subscribed = True
-        log.info(
-            "Renewed subscription to %s, sid: %s",
-            self.service.base_url + self.service.event_subscription_url,
-            self.sid)
+            if success:
+                success(response.headers)
+            return
+
+        if failure:
+            failure()
 
-    def unsubscribe(self):
-        """Unsubscribe from the service's events.
 
-        Once unsubscribed, a Subscription instance should not be reused
+class Subscriptions(object):
```

Comment:
`Subscriptions` doesn't seem like a good name to me, since it's easily misread as `Subscription` and doesn't make it clear that it works as a mapping. I'd propose `SubscriptionsMap'.